lib/resty/dns/resolver.lua

```diff
@@ -720,6 +750,32 @@ local function _tcp_query(self, query, id)
     return answers
 end
 
+local function _filter_sections(answers, opts)
+    local result = {}
+    local authority_section = false
+    local additional_section = false
+
+    if answers.errcode then
+      return answers
+    end
+
+    if opts then
+        authority_section = opts.authority_section
+        additional_section = opts.additional_section
+    end
+
+    for i, ans in ipairs(answers) do
+        if ans.section == SECTION_AN then
+           table.insert(result, ans)
+        elseif ans.section == SECTION_NS and authority_section then
+          table.insert(result, ans)
+        elseif ans.section == SECTION_AR and additional_section then
+          table.insert(result, ans)
+        end
+    end
```

Comment:
Hi @doujiang24, just get back to the code and did a double check. When we start doing `_filter` answers, the answer is already becomes a list of table and doesn't specify how many answers in each section. I guess if we want to pass the "how many items in each section" we need to change
the shape of the answers which comes from `parse_response`. And make a nested hash table which looks like:

```lua
{
  ['answers'] = { ... },
  ['authority'] = { ... },
  ['additional'] = { ... }
}
```

Then we will be able to know how many items in each section. Otherwise it will become little bit
hacky to pre-allocate the table using `new_tab` since the state of `how many answers in each section` didn't pass around in the original design.

Cheers